### PR TITLE
improve Server and Blitz Javadoc

### PR DESCRIPTION
**Pull request by mtbc**
This PR should improve the Javadoc within certain packages. Building the Javadoc should report no warnings within those classes. They are:

* `ome.api.*`
* `ome.formats.*`
* `ome.logic.*`

--no-rebase

### CODE REVIEW

File: components/blitz/src/ome/formats/importer/transfers/AbstractExecFileTransfer.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
     /**
      * "Transfer" files by soft-linking them into place. This method is likely
      * re-usable for other general "linking" strategies by overriding
-     * {@link #createProcessBuilder(File, File)} and the other protected methods here.
+     * {@code createProcessBuilder(File, File)} and the other protected methods here.
```
Comment (sbesson) [inline components/blitz/src/ome/formats/importer/transfers/AbstractExecFileTransfer.java]: Was this link causing a Javadoc exception?

File: components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 /**
  * Base {@link FileTransfer} implementation primarily providing the
- * {@link #start(TransferState)} and {@link #finish(TransferState, long)}
+ * {@code start(TransferState)} and {@code finish(TransferState, long)}
```
Comment (sbesson) [inline components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java]: Same comment as above.

File: components/blitz/src/ome/formats/importer/transfers/HardlinkFileTransfer.java [hunk 3]
```diff
@@ -41,8 +41,9 @@
      * @param file File to be copied.
      * @param location Location to copy to.
      * @throws IOException
-     * @TODO Java7: replace ln once Java6 is dropped
+     *
      */
+    // TODO Java7: replace ln once Java6 is dropped
```
Comment (sbesson) [inline components/blitz/src/ome/formats/importer/transfers/HardlinkFileTransfer.java]: Side-note: now Java6 is dropped on this line, we might want to make sure this is captured in a card for 5.2.0

File: components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java [hunk 4]
```diff
@@ -159,7 +160,7 @@ protected void printLine() {
     }
 
     /**
-     * Method used by subclasses during {@link #afterTransfer(int, List<String>)}
+     * Method used by subclasses during {@link ome.formats.importer.transfers.FileTransfer#afterTransfer(int, List)}
```
Comment (sbesson) [inline components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java]: Very minor. `{@link FileTransfer#afterTransfer(int, List)}` might be sufficient here.

File: components/blitz/src/ome/formats/importer/util/ErrorHandler.java [hunk 5]
```diff
@@ -46,8 +46,8 @@
 
 /**
  * Top of the error handling hierarchy. Will add errors to a queue
- * which can be sent with {@link #sendErrors()}. Subclasses will get
- * a change to handle all {@link ImportEvent} instances, but should
+ * which can be sent with {@code sendErrors()}. Subclasses will get
```
Comment (sbesson) [inline components/blitz/src/ome/formats/importer/util/ErrorHandler.java]: Here as well, does `@link #sendErrors()` throw a warning?

**Comment by sbesson** [comment]
Few minor comments. Otherwise, this reads well and the Javadoc renders correctly at https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/javadoc/.

**Comment by jburel** [comment]
For the modified files, we should also remove the 
```
//java imports
```

**Comment by mtbc** [comment]
@jburel: Okay if we do imports in separate PRs? They're full of conflict fun. Probably I can look for some automated help to order them properly and remove unused ones: I'll investigate.

File: components/blitz/src/ome/formats/importer/transfers/AbstractExecFileTransfer.java (hunk 1 above)
Comment (mtbc) [inline components/blitz/src/ome/formats/importer/transfers/AbstractExecFileTransfer.java]: No, something else in the file was, then I noticed it was causing a warning in Eclipse.

File: components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java (hunk 2 above)
Comment (mtbc) [inline components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java]: Same answer. :smiley:

File: components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java (hunk 4 above)
Comment (mtbc) [inline components/blitz/src/ome/formats/importer/transfers/AbstractFileTransfer.java]: I think you're right. I was puzzled over this myself. (Sometimes after some changes Eclipse gets confused and needs a kick.)

File: components/blitz/src/ome/formats/importer/transfers/HardlinkFileTransfer.java (hunk 3 above)
Comment (mtbc) [inline components/blitz/src/ome/formats/importer/transfers/HardlinkFileTransfer.java]: https://trello.com/c/35fbNCPN/20-fix-ln-in-transfer-methods

File: components/blitz/src/ome/formats/importer/transfers/AbstractExecFileTransfer.java (hunk 1 above)
Comment (mtbc) [inline components/blitz/src/ome/formats/importer/transfers/AbstractExecFileTransfer.java]: Aha. I think I had Eclipse set wrongly to not think protected methods are reported in Javadoc. I'll revert this accordingly.

**Comment by jburel** [comment]
@mtbc: if it makes things easier, of course.
We removed them in the other "cleaning" PRs and did not generate too many conflicts.

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-DEV-merge-push#76](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-push/76/). See the [console output](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-push/76/consoleText) for more details.
Possible conflicts:
  - PR #3910 joshmoore 'Add wrapper class for loading permission info (Fix #12474)'
    - components/server/src/ome/logic/QueryImpl.java

**Comment by mtbc** [comment]
Reverted change and rewrote history to avoid merge conflict with earlier PR.

**Comment by mtbc** [comment]
@jburel: Started out `//Java imports` and other such cleaning in #4115.

File: components/server/src/ome/api/local/LocalCompress.java [hunk 6]
```diff
@@ -39,15 +39,14 @@ void compressToStream(BufferedImage image, OutputStream outputStream)
 	 * 
 	 * @param percentage A percentage compression level from 1.00 (100%) to 
 	 * 0.01 (1%).
-	 * @throws ValidationException if the <code>percentage</code> is out of
-	 * range.
+	 * @throws ome.conditions.ValidationException if the {@code percentage} is out of
```
Comment (sbesson) [inline components/server/src/ome/api/local/LocalCompress.java]: The `<i>stream<i>` in the previous method lines above alters the rendering of this [page](https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/javadoc/ome/api/local/LocalCompress.html). Not the fault of this PR though - can be fixed in a follow-up PR.

**Comment by sbesson** [comment]
All the changes made in this PR make sense and are currently rendered by the Javadoc produced in the merge build. Merging so that we can keep cleaning Javadoc warnings.

**Pull request merged**